account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/Stampler.java

```diff
@@ -0,0 +1,211 @@
+    ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+    return ow.writeValueAsString(apiStamp);
+  }
+
+  private TEKManager tekManager;
+
+  private char[] bundlePrivateKey;
+  private char[] bundlePublicKey;
+  public Stampler(TEKManager tekManager) {
+    this.tekManager = tekManager;
+    this.bundlePublicKey = null;
+    this.bundlePrivateKey = null;
+  }
+
+  /**
+   * Creates a new TEK via HpkeTEKManager or returns the existing one
+   * @return TEK public key
+   */
+  public String initTek(){
+      byte[] tekPublicKeyBytes = this.tekManager.createTEK().getPublicKeyBytes().toByteArray();
+      return Utilities.bytesToHex(tekPublicKeyBytes);
+
+  }
+
+
+  /** Injects a credential bundle (Base58-encoded) after decrypting.
+   * @param bundle Base58-encoded credential bundle
+   * @throws GeneralSecurityException iif initTek is never called.
+   */
+  public void injectCredentialBundle(String bundle) throws GeneralSecurityException {
+    // In Kotlin: val tekPublicKey = tekManager.publicKey() ?: throw StamperNotInitializedException()
+    if (tekManager.getPublicKey() == null) {
+      throw new StamperNotInitializedException();
+    }
+    // TEK public key in raw bytes
+    byte[] tekPublicKeyBytes = tekManager.getPublicKey().getPublicKeyBytes().toByteArray();
+
+    // Decode bundle from Base58 (with checksum)
+    byte[] decodedBundle = Base58.decodeChecked(bundle);
+    ByteBuffer buffer = ByteBuffer.wrap(decodedBundle);
+
+    // ephemeralPublicKey (33 bytes, compressed)
+    int ephemeralPublicKeyLength = 33;
+    byte[] ephemeralPublicKeyBytes = new byte[ephemeralPublicKeyLength];
+    buffer.get(ephemeralPublicKeyBytes);
+
+    // Convert ephemeral public key into uncompressed form
+    byte [] ephemeralPublicKeyUncompressed = EllipticCurves.getEcPublicKey(
+        EllipticCurves.CurveType.NIST_P256,
+        EllipticCurves.PointFormatType.UNCOMPRESSED,
+        ephemeralPublicKeyBytes
+    ).getEncoded();
+
+
+    // Remaining bytes are the ciphertext
+    byte[] ciphertext = new byte[buffer.remaining()];
+    buffer.get(ciphertext);
+
+    // The AAD is ephemeralPublicKeyUncompressed + TEK public key
+    byte[] aad = Bytes.concat(ephemeralPublicKeyUncompressed, tekPublicKeyBytes);
+
+    // Perform HPKE decryption
+    byte[] decryptedKey = tekManager.hpkeDecrypt(
+        ephemeralPublicKeyUncompressed,
+        ciphertext,
+        "turnkey_hpke".getBytes(),
+        aad
+    );
+
+    // Split out the privateKey/publicKey from the decrypted bytes
+    byte[][] keyPair = privateKeyToKeyPair(decryptedKey);
+    byte[] publicKeyBytes = keyPair[0];
+    byte[] privateKeyBytes = keyPair[1];
+
+    // Store them, as hex in lowercase
+    this.bundlePrivateKey = Utilities.bytesToHex(privateKeyBytes).toLowerCase(Locale.ROOT).toCharArray();
+    this.bundlePublicKey = Utilities.bytesToHex(publicKeyBytes).toLowerCase(Locale.ROOT).toCharArray();
```

Comment:
is it more convenient / efficient to just skip this conversion and store the keys as `byte[]` in this class?